apps/files/models.py

```diff
@@ -378,6 +378,22 @@ def get_localepicker(self):
         statsd.timing('files.extract.localepicker', (end * 1000))
         return res
 
+    @property
+    def is_signed(self):
+        """Return True if the file has been signed.
+
+        This will simply check the signature filenames, and assume that if
+        they're named "mozilla.*" then the xpi has been signed by us.
+
+        This is in no way a perfect or correct solution, it's just the way we
+        do it until we decide to inspect/walk the certificates chain to
+        validate it comes from Mozilla."""
+        zf = zipfile.ZipFile(self.file_path, mode='r')
+        filenames = zf.namelist()
+        return ('META-INF/mozilla.rsa' in filenames and
+                'META-INF/mozilla.sf' in filenames and
+                'META-INF/manifest.mf' in filenames)
```

Comment:
I don't think we need to explicitely close the file here (because zf will get out of scope at the end of the method, and will be automatically closed).

However, now that we've switch to using py27, we can start using the context manager form of zipfile.Zipfile, which is much cleaner. I'll do this modification, and use sets as you advised.